FakeServer/GraphQL/GraphQLMiddleware.cs

```diff
@@ -44,25 +46,39 @@ public GraphQLMiddleware(RequestDelegate next, IDataStore datastore, IMessageBus
                 return;
             }
 
-            if (context.Request.Method != "POST" || !context.Request.ContentType.Contains("application/graphql"))
+            if (context.Request.Method != "POST" ||
+                !context.Request.ContentType.ContainsAny("application/graphql", "application/json"))
             {
                 context.Response.StatusCode = (int)HttpStatusCode.NotImplemented;
                 await context.Response.WriteAsync(JsonConvert.SerializeObject(new { errors = new[] { "Not implemented" } }));
                 return;
             }
 
-            var query = string.Empty;
-
-            using (var streamReader = new StreamReader(context.Request.Body))
+            var result = default(GraphQLResult);
+            try
             {
-                query = await streamReader.ReadToEndAsync().ConfigureAwait(true);
-            }
+                string query = await ParseQuery(context);
 
-            var toReplace = new[] { "\r\n", "\\r\\n", "\\n", "\n" };
+                var toReplace = new[] { "\r\n", "\\r\\n", "\\n", "\n" };
 
-            query = toReplace.Aggregate(query, (acc, curr) => acc.Replace(curr, ""));
+                query = toReplace.Aggregate(query, (acc, curr) => acc.Replace(curr, ""));
+
+                result = await GraphQL.HandleQuery(query, _datastore);
+            }
+            catch (Exception e)
+            {
+                if (result == default(GraphQLResult))
+                {
+                    result = new GraphQLResult() { Errors = new List<string>() };
+                }
+                else if (result.Errors == default(List<string>))
+                {
+                    result.Errors = new List<string>();
+                }
 
-            var result = await GraphQL.HandleQuery(query, _datastore);
+                result.Errors.Add(e.Message);
+
```

Comment:
There was some empty lines etc. I recommed to use some extension e.g. Code maid for code cleanup
http://www.codemaid.net/documentation/#cleaning